sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -961,13 +961,15 @@ object TypeCoercion extends TypeCoercionBase {
 
       // Implicit cast between date time types
       case (DateType, TimestampType) => TimestampType
+      case (DateType, AnyTimestampType) => TimestampType
```

Comment:
here should be `AnyTimestampType.defaultConcreteType` as well

Reply:
I will address this in the PR for extracting date fields.